dkg-gadget/src/worker.rs

```diff
@@ -492,15 +595,34 @@ where
 	) {
 		debug!(target: "dkg", "🕸️  Process DKG message {}", &dkg_msg);
 
-		match self.rounds.handle_incoming(dkg_msg.payload) {
-			Ok(()) => (),
-			Err(err) => debug!(target: "dkg", "🕸️  Error while handling DKG message {:?}", err),
-		}
-		self.send_outgoing_dkg_messages();
+		if self.current_validator_set.id == GENESIS_AUTHORITY_SET_ID {
```

Comment:
Same as above, we may receive incoming dkg message for any of the rounds or next_rounds. I think we can pass the same message to both of the rounds instances and let them decide whether they're interested in that particular message. (We'll need to add additional checks in MultiPartyECDSARounds::vote() method)